kinto/core/resource/schema.py

```diff
@@ -213,11 +215,15 @@ class CollectionQuerySchema(QuerySchema):
     _limit = QueryField(colander.Integer())
```

Comment:
Also, with the current implementation we also accept negative values, which can overflow as well.